base_decorator.rb

```diff
@@ -0,0 +1,30 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Nameable
```

Comment:
- [ ] The class CapitalizeDecorator should inherit from BaseDecorator, Not Nameable class, kindly fix this so your Decorator works as expected

Reply:
Thank you very much